editor/src/messages/portfolio/document/utility_types/network_interface.rs

```diff
@@ -453,114 +454,143 @@ impl NodeNetworkInterface {
 		}
 	}
 
-	/// Get the [`Type`] for any InputConnector
-	pub fn input_type(&self, input_connector: &InputConnector, network_path: &[NodeId]) -> Type {
-		// TODO: If the input_connector is a NodeInput::Value, return the type of the tagged value
-		let node_type_from_compiled_network = if let Some(node_id) = input_connector.node_id() {
-			let Some(current_network) = self.network(network_path) else {
-				log::error!("Could not get current network in input_type");
-				return concrete!(());
-			};
-			let Some(node) = current_network.nodes.get(&node_id) else {
-				log::error!("Could not get node {node_id} in input_type");
-				return concrete!(());
-			};
-			let node_id_path = [network_path, &[node_id]].concat().clone();
-			match &node.implementation {
-				DocumentNodeImplementation::Network(nested_network) => {
-					let downstream_connection = nested_network
-						.nodes
-						.iter()
-						.flat_map(|(node_id, node)| node.inputs.iter().enumerate().map(|(input_index, input)| (InputConnector::node(*node_id, input_index), input)))
-						.chain(nested_network.exports.iter().enumerate().map(|(export_index, export)| (InputConnector::Export(export_index), export)))
-						.find(|(_, input)| {
-							if let NodeInput::Network { import_index, .. } = input {
-								*import_index == input_connector.input_index()
-							} else {
-								false
-							}
-						});
-					if let Some((input_connector, _)) = downstream_connection {
-						Some(self.input_type(&input_connector, &node_id_path))
-					}
-					// Nothing is connected to the import
-					else {
-						Some(concrete!(()))
+	/// Try and get the [`DocumentNodeDefinition`] for a node
+	pub fn get_node_definition(&self, network_path: &[NodeId], node_id: NodeId) -> Option<&DocumentNodeDefinition> {
```

Comment:
Im not sure how True Doctors PR to automatically generate definitions for all nodes fits into this, but this function will also have to return those definitions.

Reply:
If you mean #1942, then that will be fine since the node definitions are added to the list used by the function here. Therefore no change will be necessary.